### PR TITLE
Exposed `options` on Markdown

### PR DESCRIPTION
**Pull request by edumoreira1506**
#### What does this PR do?
This PR adds a `fill` prop to markdown component.

Resolves issue #3568

### CODE REVIEW

**Comment by ShimiSun** [comment]
Hi @edumoreira1506, thank you for picking up this issue and. helping us start a conversation here.

After looking into it,  and reviewing https://github.com/probablyup/markdown-to-jsx/blob/master/README.md#optionsforceinline, I think the user should be allowed to set the `forceInLine` prop via `options` and not as a new grommet prop.
This conclusion leads me to a few questions:
1. Why aren't we exposing the usage of the React Markdown component 'markdown-to-jsx',  if the user will know it, it could be simpler to research.
2. why aren't we exposing the `options` prop? it is there to use but we are not exposing it on the docs. 
I was able to solve the issue this PR is trying to solve only adding `options={{ forceInline: true }}` to the Markdown thanks to the `options` prop and the knowledge of markdown-to-jsx.

thoughts here @ericsoderberghp ? @oorestisime ?

**Comment by edumoreira1506** [comment]
Maybe I could change the docs to show for user than we are using `markdown-to-jsx`.

**Comment by oorestisime** [comment]
yeah i agree it makes sense to add the options in the docs. and mention we use that component so they will know what to search

**Comment by ericsoderberghp** [comment]
I'm supportive of adding `fill` to Markdown as it seems consistent with the other grommet components that have `fill`.

**Comment by ericsoderberghp** [comment]
After an excellent discussion with @ShimiSun, I am withdrawing my support for adding `fill` to Markdown. As, this isn't filling anything you might have in Markdown, such as an Image. It's just changing the behavior of Paragraph. For that, I think @oorestisime's suggestion to customize Paragraph props is the right approach. However, I am supportive of documenting the `options` property.

**Comment by ShimiSun** [comment]
@edumoreira1506 , after applying the changes to doc.js, please run locally `yarn test-update` so it will create updated snapshots with the new content, push those snapshots as part of the PR and it will resolve the jest failures.

File: src/js/components/Markdown/README.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 ## GrommetMarkdown
 Markdown formatting using Grommet components.
 
+Grommet uses 'markdown-to-jsx' in Markdown component,
+      you can see all the options in the documentation.
```
Comment (ShimiSun) [inline src/js/components/Markdown/README.md]: thanks for adding this, can you also add the new prop of `options` (similarly to `components` prop) and direct to 'markdown-to-jsx' with a link?

Comment (edumoreira1506) [reply src/js/components/Markdown/README.md]: Yes, sure!

**Comment by edumoreira1506** [comment]
Updated.

File: src/js/components/Markdown/README.md [hunk 2]
```diff
@@ -17,6 +20,10 @@ Custom components and props to override html elements such as 'img'
       or 'pre'. By default 'a', 'p', 'img', and table elements are overridden
       with grommet components
 
+**options**
+
+You can see all the options [here](https://github.com/probablyup/markdown-to-jsx).
+
```
Comment (ShimiSun) [inline src/js/components/Markdown/README.md]: The README file is auto-generated when running `yarn test-update` and it shouldn't be manually edited.
A way to achieve it is to add `options` prop on the doc.js file and then auto-generate the readme to include the new prop details.

**Comment by ShimiSun** [comment]
Thank you for your contribution!

**Pull request merged**